algebird-core/src/main/scala/com/twitter/algebird/Interval.scala

```diff
@@ -22,31 +22,31 @@ package com.twitter.algebird
  * Represents a single interval on a T with an Ordering
  */
 sealed trait Interval[T] extends java.io.Serializable {
-  def contains(t: T): Boolean
+  def contains(t: T)(implicit ord: Ordering[T]): Boolean
 
-  def intersect(that: Interval[T]): Interval[T]
-  def apply(t: T) = contains(t)
-  def &&(that: Interval[T]) = intersect(that)
+  def intersect(that: Interval[T])(implicit ord: Ordering[T]): Interval[T]
+  def apply(t: T)(implicit ord: Ordering[T]) = contains(t)
+  def &&(that: Interval[T])(implicit ord: Ordering[T]) = intersect(that)
 
   /**
    * Map the Interval with a non-decreasing function.
    * If you use a non-monotonic function (like x^2)
    * then the result is meaningless.
    * TODO: It might be good to have types for these properties in algebird.
    */
-  def mapNonDecreasing[U: Ordering](fn: T => U): Interval[U]
+  def mapNonDecreasing[U](fn: T => U): Interval[U]
```

Comment:
So a law might be: forAll intervals and values t, if we map t and the interval, contains returns the same value.

To make the function non-decreasing we will need to do something like make a new fn and apple the first and if the result is less return the input.